### PR TITLE
Add visualization output via tensorboard to the clustering example

### PR DESCRIPTION
**Pull request by benkli01**
**Note**: Batch-wise summaries of the clustering can be slow and might lead to a warning like this:

>     "WARNING:tensorflow:Method (on_train_batch_end) is slow compared
>      to the batch update (0.102097). Check your callbacks."

### CODE REVIEW

File: tensorflow_model_optimization/python/core/clustering/keras/BUILD
Comment (akarmi) [inline tensorflow_model_optimization/python/core/clustering/keras/BUILD:80]: Please, change to "PY3" - we now only support Python3 in TFMOT.

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py
Comment (akarmi) [inline tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py:11]: nit: "histograms"

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py
Comment (akarmi) [inline tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py:24]: Should we validate log_dir here, similar to how it's done in PruningSummaries?

Comment (benkli01) [reply tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py:24]: We could, but I would leave that to the base class and/or the filewriter as they are actually using the `log_dir` directly. What do you think?

Comment (akarmi) [reply tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py:24]: I would agree, however, I would also try to make the behaviour of all APIs in MOT similar wherever possible. I guess we at least should make sure that passing invalid log_dir value would cause similar behaviour of __init__ in "summary" callback in both APIs. Do you know if tf.summary calls below raise an exception in such a case?

Comment (alanchiao) [reply tensorflow_model_optimization/python/core/clustering/keras/clustering_callbacks.py:24]: Just as a pass-by, like Anton, I would also agree that the base class should be responsible. It happens to be that there is no error message today though and some sort of silent failure (I think maybe it just doesn't create a file - you can check), which a user ran into. Until we can make the base classes error well, this is useful.

**Review by akarmi — APPROVED** [review]
LGTM. Thanks.

**Comment by akarmi** [comment]
@benkli01, please resolve merge conflict

**Comment by benkli01** [comment]
Merge conflict resolved and commit message adapted, because the simplification of the example (#516) is already merged.

**Comment by alanchiao** [comment]
The commit has been added to Github (see [here](https://github.com/tensorflow/model-optimization/commit/5603678fd8dd5288e20ca11f734dd21b9541ab08)) and attributed to Benjamin. Unfortunately, due to the hiccup, I don't think we can ever mark the PR itself as merged.

The cause was that I went through the regular merge procedure, but there was an issue that showed up after merging, from an internal check that makes sure that there is the "COPYRIGHT" part at the top of every file. I'm changing things so that the check will run earlier during "import/copybara" and this issue won't arise again. Apologize for the trouble.

**Comment by alanchiao** [comment]
Closing this given the above comment. The merged commit is here: https://github.com/tensorflow/model-optimization/commit/5603678fd8dd5288e20ca11f734dd21b9541ab08.

**Comment by benkli01** [comment]
Thanks for taking care of this, @alanchiao !

**Comment by benkli01** [comment]
Some of the latest changes of this PR seem to be missing in the commit to the master branch. I will have a closer look and, if needed, I will just create a new PR to amend this.

**Pull request closed without merging**
